cli/src/main.rs

```diff
@@ -563,7 +563,7 @@ impl PathHash {
 /// Read a file.
 fn read(path: &Path) -> FileResult<Vec<u8>> {
     let f = |e| FileError::from_io(e, path);
-    if fs::metadata(&path).map_err(f)?.is_file() {
```

Comment:
I think it would be clearer to flip the condition and make an early return here. Could you also update the similar piece of code in `tests/src/tests.rs`?

Reply:
Done.